File: domain/src/main/java/com/krillsson/sysapi/domain/gpu/GpuLoad.java
package com.krillsson.sysapi.domain.gpu;

import com.fasterxml.jackson.annotation.JsonProperty;

public class GpuLoad
{
    private double temperature;
    private double coreLoad;
    private double memoryLoad;

    public GpuLoad(double temperature, double coreLoad, double memoryLoad)
    {
        this.temperature = temperature;
        this.coreLoad = coreLoad;
        this.memoryLoad = memoryLoad;
    }

    @JsonProperty
    public double getTemperature()
    {
        return temperature;
    }

    @JsonProperty
    public double getCoreLoad()
    {
        return coreLoad;
    }

    @JsonProperty
    public double getMemoryLoad()
    {
        return memoryLoad;
    }
}

File: server/src/main/java/com/krillsson/sysapi/core/history/db/GpuLoad.kt
package com.krillsson.sysapi.core.history.db

import java.util.*
import javax.persistence.*

@Entity
class GpuLoad(
    @Id
    val id: UUID,
    @JoinColumn(name = "historyId", insertable = false, updatable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    val history: HistorySystemLoadEntity? = null,
    val historyId: UUID,
    val name: String,
    val coreLoad: Double,
    val memoryLoad: Double,
    @Embedded
    val health: GpuHealth
)

@Embeddable
class GpuHealth(val fanRpm: Double, val fanPercent: Double, val temperature: Double)